src/sentry/tagstore/legacy/backend.py

```diff
@@ -0,0 +1,144 @@
+"""
+sentry.tagstore.legacy.backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2017 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+
+from __future__ import absolute_import
+
+import six
+
+from django.db.models import Q
+from operator import or_
+from six.moves import reduce
+
+from sentry import buffer
+from sentry.tagstore import TagKeyStatus
+from sentry.models import TagKey, TagValue, EventTag
```

Comment:
As a follow up iteration, it'd be nice to move these models under `sentry.tagstore.legacy.models.*` namespace to isolate them out of the main list of models. Similar to how `nodestore` is structured. That further reinforces that these models shouldn't be used directly outside of `tagstore`.

Reply:
Does that change the Django app it's part of, or would that still be `sentry`? I guess I need to just try it. For some reason I was thinking moving *existing* models (I think `nodestore` were net-new) would break a lot of shit.

Reply:
I definitely would suggest it as an isolated change, but it shouldn't be a problem as long as the table names are explicitly defined, which we more often than not do. So we can't change the names of tables, but we can just easily change their import path. New tables if we need them can then just be like, `tagstore_tagkey` or whatever.